core/src/main/java/org/owasp/dependencycheck/data/update/nvd/api/NvdApiProcessor.java

```diff
@@ -100,13 +98,32 @@ public NvdApiProcessor call() throws Exception {
                     LOGGER.error("Failed to process " + entry.getCve().getId(), ex);
                 }
             }
-        } finally {
-            itemSource.close();
         }
         endTime = System.currentTimeMillis();
         return this;
     }
 
+    static CveItemSource<DefCveItem> buildItemSource(File file) throws IOException {
+        if (file.getName().endsWith(".jsonarray.gz")) {
+            try (InputStream fis = Files.newInputStream(file.toPath());
+                 InputStream zis = new GZIPInputStream(fis);
+                 InputStream is = new BufferedInputStream(zis)) {
+                return new JsonArrayCveItemSource(is);
+            }
+        } else if (file.getName().endsWith(".gz")) {
+            try (InputStream fis = Files.newInputStream(file.toPath());
+                 InputStream zis = new GZIPInputStream(fis);
+                 InputStream is = new BufferedInputStream(zis)) {
+                return new CveApiJson20CveItemSource(is);
+            }
+        } else {
+            try (InputStream fis = Files.newInputStream(file.toPath());
+                 InputStream is = new BufferedInputStream(fis)) {
+                return new JsonArrayCveItemSource(is);
+            }
```

Comment:
These will get closed in the ItemSource `close()`:

```suggestion
            InputStream fis = Files.newInputStream(file.toPath());
            InputStream zis = new GZIPInputStream(fis);
            InputStream is = new BufferedInputStream(zis));
            return new JsonArrayCveItemSource(is);
        } else if (file.getName().endsWith(".gz")) {
            InputStream fis = Files.newInputStream(file.toPath();
            InputStream zis = new GZIPInputStream(fis);
            InputStream is = new BufferedInputStream(zis));
            return new CveApiJson20CveItemSource(is);
        } else {
            InputStream fis = Files.newInputStream(file.toPath();
            InputStream is = new BufferedInputStream(fis));
            return new JsonArrayCveItemSource(is);
```

Reply:
already pushing